src/Microsoft.AspNetCore.Hosting/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -0,0 +1,280 @@
+            var startTimestamp = (diagnoticsEnabled || loggingEnabled) ? Stopwatch.GetTimestamp() : 0;
+
+            // Scope may be relevant for a different level of logging, so we always create it
+            // see: https://github.com/aspnet/Hosting/pull/944
+            context.Scope = HostingLoggerExtensions.RequestScope(_logger, httpContext);
+
+            if (loggingEnabled)
+            {
+                // Non-inline
+                LogRequestStarting(httpContext);
+            }
+#if !NET451
+            if (_diagnosticSource.IsEnabled(ActivityName, httpContext))
+            {
+                context.Activity = StartActivity(httpContext);
+            }
+#endif
+            if (diagnoticsEnabled)
+            {
+                // Non-inline
+                RecordBeginRequestDiagnostics(httpContext, startTimestamp);
+            }
+
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void RequestEnd(HttpContext httpContext, Exception exception, HostingApplication.Context context)
+        {
+            // Local cache items resolved multiple items, in order of use so they are primed in cpu pipeline when used
+            var startTimestamp = context.StartTimestamp;
+
+            // If startTimestamp is 0, don't call GetTimestamp, likely don't need the value
+            var currentTimestamp = (startTimestamp != 0) ? Stopwatch.GetTimestamp() : 0;
+
+            // To keep the hot path short we defer logging to non-inlines
+            if (exception == null)
+            {
+                // No exception was thrown, request was sucessful
+                if (_diagnosticSource.IsEnabled(DiagnosticsEndRequestKey))
+                {
+                    // Diagnostics is enabled for EndRequest, but it may not be for BeginRequest
+                    // so call GetTimestamp if currentTimestamp is zero (from above)
+                    RecordEndRequestDiagnostics(
+                        httpContext,
+                        (currentTimestamp != 0) ? currentTimestamp : Stopwatch.GetTimestamp());
+                }
+            }
+            else
+            {
+                // Exception was thrown from request
+                if (_diagnosticSource.IsEnabled(DiagnosticsUnhandledExceptionKey))
+                {
+                    // Diagnostics is enabled for UnhandledException, but it may not be for BeginRequest
+                    // so call GetTimestamp if currentTimestamp is zero (from above)
+                    RecordUnhandledExceptionDiagnostics(
+                        httpContext,
+                        (currentTimestamp != 0) ? currentTimestamp : Stopwatch.GetTimestamp(),
+                        exception);
+                }
+
+                if (context.EventLogEnabled)
+                {
+                    // Non-inline
+                    HostingEventSource.Log.UnhandledException();
+                }
+            }
+
+#if !NET451
```

Comment:
Meaningless? Aren't building for `NET451` only for `netstandard1.3` and `netstandard1.5`